tensorflow/core/kernels/image/decode_image_op.cc

```diff
@@ -452,12 +452,12 @@ class DecodeImageV2Op : public OpKernel {
     // allocation til after dtype conversion is done. `gif`::Decode` supports
     // uint8 only.
     Tensor* output = nullptr;
-    int buffer_size = 0;
+    ptrdiff_t buffer_size = 0;
```

Comment:
Why `ptrdiff_t` and not `size_t`?

Reply:
@mihaimaruseac I've replaced it with int64_t now. I didn't use size_t because I didn't want to compare signed with unsigned, and ptrdiff_t is usually as large as size_t, but signed. And because EIGEN_DEFAULT_DENSE_INDEX_TYPE which is the type of the TensorMap constructor's second parameter is ptrdiff_t by default.